scripts/train_agents.py

```diff
@@ -2,16 +2,200 @@
+        args.SP_seed = args.ADV_seed
+        args.SP_h_dim = args.ADV_h_dim
+        args.learner_type = main_agent_type
+        args.reward_magnifier = reward_magnifier
+        args.exp_dir = exp_tag 
+        SP(args=args, pop_force_training=True)
+        sp_tag = 'sp_s' + args.SP_seed + '_h' + args.SP_h_dim + '_tr(SP)_ran'
+        main_agent_path = exp_tag + '/' + sp_tag
+    root = 'agent_models/' + main_agent_path
+    root_adv = root + '/' + exp_tag
+
+    adv_tag = 'adv_s' + args.ADV_seed + '_h' + args.ADV_h_dim + '_tr(H)_ran'
+    pwadv_tag = 'pwadv_s' + args.ADV_seed + '_h' + args.ADV_h_dim + '_tr(SP_SPADV)_ran'
+
+    how_long = how_long_for_adv
+    set_input(args=args, quick_test=quick_test, how_long=how_long)
+    args.learner_type = adversary_type
+    args.reward_magnifier = reward_magnifier
+    args.exp_dir = root_adv + '/' + adversary_type + '/0'
+    ADV(args=args, 
+        agent_folder_path = root, 
+        agent_file_tag= CheckedPoints.BEST_EVAL_REWARD)
+
+    how_long = how_long_init + how_long_for_agent
+    set_input(args=args, quick_test=quick_test, how_long=how_long)
+    args.learner_type = main_agent_type
+    args.reward_magnifier = reward_magnifier
+    args.exp_dir = root_adv + '/' + main_agent_type + '-' + adversary_type + 'play' + '/0'
+    PwADVs( args=args, 
+            agent_folder_path = root, 
+            agent_file_tag = CheckedPoints.BEST_EVAL_REWARD,
+            adv_folder_paths = [root_adv + '/' + adversary_type + '/0'], 
+            adv_file_tag = adv_tag + '/' + checked_adversary)
+    ###################################################################
+    for round in range(1,rounds_of_advplay):
+        how_long = how_long_for_adv
+        set_input(args=args, quick_test=quick_test, how_long=how_long)
+        args.learner_type = adversary_type
+        args.reward_magnifier = reward_magnifier
+        args.exp_dir = root_adv + '/' + adversary_type + '/' + str(round)
+        ADV(args=args,
+            agent_folder_path = root_adv + '/' + main_agent_type + '-' + adversary_type + 'play' + + str(round-1), 
+            agent_file_tag = pwadv_tag + '/' + CheckedPoints.FINAL_TRAINED_MODEL)
+        
+        how_long = how_long_init + how_long_for_agent*(round+1)
+        set_input(args=args, quick_test=quick_test, how_long=how_long)
+        args.learner_type = main_agent_type
+        args.reward_magnifier = reward_magnifier
+        args.exp_dir = root_adv + '/' + main_agent_type + '-' + adversary_type + 'play' + '/' + str(round)
+        PwADVs( args=args, 
+                agent_folder_path = root, 
+                agent_file_tag = CheckedPoints.BEST_EVAL_REWARD,
+                adv_folder_paths = [f"{root_adv}/{adversary_type}/{round_num}" for round_num in range(round)], 
+                adv_file_tag = adv_tag + '/' + checked_adversary)
+
+def PwADVs(args, 
+          agent_folder_path='agent_models/four-layouts/supporter/0', 
```

Comment:
The default values for these inputs are magic.

Reply:
That's a clear mistake here. I need to rewrite it.